src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -129,11 +136,21 @@ const CustomFieldInputDate = forwardRef(function CustomFieldInputDate(props, ref
     setCurrentValue(event.target.value);
   };
 
+  const toLocaleDate = (date) => {
+    return date.toLocaleDateString(undefined, { month: 'short', year: 'numeric', day: 'numeric' });
+  };
+
   const errorText = () => {
     if (!isValid && !isValidInput(props.value)) {
       return `"${props.value}" is an invalid date`;
     }
 
+    const dateMilliseconds = Date.parse(currentValue);
+    const date = new Date(dateMilliseconds);
+
+    if (date > maxDate || date < minDate) {
+      return `Date must be between ${toLocaleDate(minDate)} and ${toLocaleDate(maxDate)}`;
```

Comment:
We will want to leverage the native validation messages. Otherwise, we'll end up in a weird spot for localization. 

What are the validation messages with `min` / `max` dates?

Reply:
After a little deliberation we are going to move to a solution to what Ben posted in MDS slack. We will just manage the error text in the parent component for now and probably close this PR.